www-apps/gitea/gitea-1.9.4.ebuild

```diff
@@ -0,0 +1,116 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+inherit golang-vcs-snapshot tmpfiles systemd
+
+EGO_PN="code.gitea.io/gitea"
+
+DESCRIPTION="A painless self-hosted Git service"
+HOMEPAGE="https://gitea.io"
+SRC_URI="https://github.com/go-gitea/gitea/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT Apache-2.0 BSD MPL-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64"
+IUSE="+acct pam sqlite"
+
+BDEPEND="<dev-lang/go-1.13"
+COMMON_DEPEND="
+	acct? (
+		acct-group/git
+		acct-user/git[gitea]
+	)
+	pam? ( sys-libs/pam )"
+DEPEND="${COMMON_DEPEND}"
+RDEPEND="${COMMON_DEPEND}
+	dev-vcs/git
+	acct? (
+		!dev-vcs/gitolite
+		!dev-vcs/gitolite-gentoo
+	)
+	"
+
+DOCS=( custom/conf/app.ini.sample CONTRIBUTING.md README.md )
+S="${WORKDIR}/${P}/src/${EGO_PN}"
+
+PATCHES=( "${FILESDIR}/gitea-mod-vendor.patch" "${FILESDIR}/gitea-logflags.patch" )
+
+gitea_make() {
+	local my_tags=(
+		bindata
+		$(usev pam)
+		$(usex sqlite 'sqlite sqlite_unlock_notify' '')
+	)
+	local my_makeopt=(
+		DRONE_TAG=${PV}
+		TAGS="${my_tags[@]}"
+		LDFLAGS="-extldflags \"${LDFLAGS}\""
+	)
+	GOPATH=${WORKDIR}/${P}:$(get_golibdir_gopath) emake "${my_makeopt[@]}" "$@"
+}
+
+src_prepare() {
+	default
+
+	local sedcmds=(
+		-e "s#^RUN_MODE = dev#RUN_MODE = prod#"
+		-e "s#^ROOT =#ROOT = ${EPREFIX}/var/lib/gitea/gitea-repositories#"
+		-e "s#^ROOT_PATH =#ROOT_PATH = ${EPREFIX}/var/log/gitea#"
+		-e "s#^APP_DATA_PATH = data#APP_DATA_PATH = ${EPREFIX}/var/lib/gitea/data#"
+		-e "s#^HTTP_ADDR = 0.0.0.0#HTTP_ADDR = 127.0.0.1#"
+		-e "s#^MODE = console#MODE = file#"
+		-e "s#^LEVEL = Trace#LEVEL = Info#"
+		-e "s#^LOG_SQL = true#LOG_SQL = false#"
+		-e "s#^DISABLE_ROUTER_LOG = false#DISABLE_ROUTER_LOG = true#"
+		-e "s#^APP_ID =#;APP_ID =#"
+		-e "s#^TRUSTED_FACETS =#;TRUSTED_FACETS =#"
+	)
+
+	sed -i "${sedcmds[@]}" custom/conf/app.ini.sample || die
+	if use sqlite ; then
+		sed -i -e "s#^DB_TYPE = .*#DB_TYPE = sqlite3#" custom/conf/app.ini.sample || die
+	fi
+
+	gitea_make generate
+}
+
+src_compile() {
+	gitea_make build
+}
+
+src_install() {
+	dobin gitea
+
+	einstalldocs
+
+	newconfd "${FILESDIR}/gitea.confd-r1" gitea
+	newinitd "${FILESDIR}/gitea.initd-r3" gitea
+	newtmpfiles - gitea.conf <<-EOF
+		d /run/gitea 0755 git git
```

Comment:
Will this work even with `USE="-acct"`? 

Also I dont think the acct is very descriptive but cant really come up with anything better either :\

Reply:
this only has an effect, when using gitea via unix sockets instead of tcp sockets.
if you change the user (`USE="-acct"`- no accounting done by portage, was the best I could find - actually, in my opinion it would be nice to move this to the accouting eclass, but anyway... ), then you can use `/usr/lib/tmpfiles.d/gitea.conf` as a template and copy it to `/etc/tmpfiles.d`  + fixing the user name.
we could also install it to `/etc/tmpfiles.d` directly (unfortunately the tmp-files eclass does not support this - so the functionality from there`insinto/newins/tmpfiles --create` has to be moved to the ebuild).

maybe the best we could do is to **remove this package entirely**, because the gentoo-proxymainter project doesn't work very well and is to slow to keep track with upstream release.
furthermore the vendor license issue is still pending - i'm not confident that this PR will get merged before `gitea-1.10` is released. I get more and more the impression that gentoo Q/A stuff, exactly does the opposite of what it should be meant to do.

Reply:
I'm tempted to just leave the remaining license issue for a bug report and to be handled later on. One more question: would `go-module.eclass` and using package *get-ego-vendor* help with the sandbox issues when building?

Reply:
you mean *get_module_vendor_uris* with *EGO_VENDOR* ?

the you would download again all vendor modules, which are actually already bundle by upstream.